multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/validators/v2/DeploymentDescriptorValidator.java

```diff
@@ -61,6 +65,23 @@ protected boolean isSupported(Resource resource) {
         return handler.findResourceType(platform, resource.getType()) != null;
     }
 
+    protected void checkForCircularDependency(Resource resource) {
+        if (hasDependentResources(resource)) {
+            for (String dependentResourceName : resource.getProcessedAfter()) {
+                Resource dependentResource = handler.findResource(descriptor, dependentResourceName);
+                if (hasDependentResources(dependentResource) && dependentResource.getProcessedAfter()
+                                                                                 .contains(resource.getName())) {
```

Comment:
This only checks the Resource's direct dependencies. It does not check the whole dependency tree.